src/lib/journey/callbacks/polling-wait/polling-wait.svelte

```diff
@@ -14,28 +14,41 @@
   import type { Maybe } from '$lib/interfaces';
 
   // Unused props. Setting to const prevents errors in console
-  export const stepMetadata: Maybe<StepMetadata> = null;
   export const style: z.infer<typeof styleSchema> = {};
 
   export let callback: PollingWaitCallback;
   export let callbackMetadata: Maybe<CallbackMetadata>;
   export let selfSubmitFunction: Maybe<SelfSubmitFunction> = null;
+  export let stepMetadata: Maybe<StepMetadata> = null;
 
   let message: string;
-
-  // Ensure this is written outside of the Reactive blog, or it get's called multiple times
-  setTimeout(() => {
-    if (callbackMetadata) { callbackMetadata.derived.isReadyForSubmission = true; }
-    selfSubmitFunction && selfSubmitFunction();
-  }, callback.getWaitTime());
+  let timer: ReturnType<typeof setTimeout>;
 
   $: {
     callback = callback as PollingWaitCallback;
     message = callback.getMessage();
+
+    // Clear any existing timeouts to avoid duplicates
+    clearTimeout(timer);
+
+    // Create new timeout, now that the previous has been cleared
+    timer = setTimeout(() => {
+      if (callbackMetadata) {
+        callbackMetadata.derived.isReadyForSubmission = true;
+      }
+      selfSubmitFunction && selfSubmitFunction();
+    }, callback.getWaitTime());
   }
 </script>
 
 <div class="tw_text-center">
-  <Spinner colorClass="tw_text-primary-light" layoutClasses="tw_h-24 tw_mb-6 tw_w-24" />
-  <Text>{message}</Text>
+  {#if stepMetadata?.derived.stageName === 'QRCode'}
```

Comment:
After reviewing this, I'm thinking of changing this like I did the `ConfirmationCallback` component.